src/main/java/com/ocient/jdbc/XGConnection.java

```diff
@@ -4984,19 +5006,20 @@ public void setPSO(final long threshold) throws Exception
 
 		try
 		{
-			out.write(intToBytes(wrapper.getSerializedSize()));
-			wrapper.writeTo(out);
-			out.flush();
-			getStandardResponse();
+			CompletionStage<ClientWireProtocol.ConfirmationResponse> completionStage = sendRequestWithStandardResponse(wrapper);
+
+			// Return CompletionStage<Void> on success
+			return completionStage.thenAccept(Response -> {});
```

Comment:
(nit):

Prefer `thenApply()` to convey mapping from `T` => `U`. In this case `U` happens to be `Void` so the net effect is mostly the same, but the intent is much clearer when using `thenApply(...)`.
```suggestion
			return sendRequestWithStandardResponse(wrapper).thenApply(ign -> null);
```